cdap-explore-client/src/main/java/co/cask/cdap/explore/client/AbstractExploreClient.java

```diff
@@ -316,17 +319,23 @@ public void run() {
                 onSuccess(handle);
               }
             }, 300, TimeUnit.MILLISECONDS);
-          } else {
-            if (!status.hasResults()) {
-              close(handle);
-            }
-            if (!resultFuture.set(new ClientExploreExecutionResult(AbstractExploreClient.this,
-                                                                   handle, status))) {
-              close(handle);
-            }
+            return;
+          }
+          if (QueryStatus.OpStatus.ERROR.equals(status.getStatus())) {
+            throw new SQLException(status.getErrorMessage(), status.getSqlState());
```

Comment:
Won't this exception get suppressed since it is getting thrown in the onSuccess method that gets executed in the executor. Should we instead set resultFuture with the exception message?